25.1/xenial/Dockerfile

```diff
@@ -0,0 +1,76 @@
+FROM ubuntu:16.04
+
+# Install dependencies
+RUN apt-get update && \
+    apt-get install -y \
+            autoconf \
+            automake \
+            build-essential \
+            curl \
+            git \
+            imagemagick \
+            ispell \
+            libdbus-1-dev \
+            libgif-dev \
+            libgnutls-dev \
+            libgtk2.0-dev \
+            libjpeg-dev \
+            libmagick++-dev \
+            libncurses-dev \
+            libpng-dev \
+            libtiff-dev \
+            libx11-dev \
+            libxpm-dev \
+            texinfo \
+            && \
+    rm -rf /var/lib/apt/lists/*
+
+# Build emacs
+COPY emacs-25.1 /tmp/emacs/
+RUN cd /tmp/emacs && \
+    ./autogen.sh && \
+    ./configure --prefix=/opt/emacs && \
+    make -j 8 install && \
+    rm -rf /tmp/emacs
+
+#--------------------------------------------------------------------------------
+
```

Comment:
Interesting, do you think there's a need tho? I will already have quite a lot of tags... If things go well the `-alpine` version will come, and if I follow your idea this will add another variant, so we'll have:

- full (with /tmp/emacs)
- normal (without /tmp/emacs)
- slim (without /tmp/emacs and without `-dev` packages)
- alpine (from alpine, without /tmp/emacs and without `-dev` packages)

If you look at https://github.com/Silex/docker-emacs/blob/refactor/README.md, the list is decently long... add `24.3` and `24.4`, then double that list because of `-alpine` and `-full`. This becomes a wall of tags :-)

Reply:
Well the other alternative is to encourage the “slim” builds, which are guaranteed to be even smaller than this. Just dev, slim and alpine.

Reply:
@gonewest818: I like your idea! tags `25.3-dev`, `25.3` and `25.3-alpine`